pdl-compiler/src/backends/rust/encoder.rs

```diff
@@ -237,15 +237,41 @@ impl Encoder {
         let shift = self.bit_shift;
 
         match &field.desc {
-            ast::FieldDesc::Flag { optional_field_id, set_value, .. } => {
-                let optional_field_id = optional_field_id.to_ident();
+            ast::FieldDesc::Flag { optional_field_id, .. } => {
+                let field_id = field.id().unwrap().to_ident();
+                let cond_field_id = optional_field_id[0].0.clone().to_ident();
+                let set_value = optional_field_id[0].1;
                 let cond_value_present =
                     syn::parse_str::<syn::LitInt>(&format!("{}", set_value)).unwrap();
                 let cond_value_absent =
                     syn::parse_str::<syn::LitInt>(&format!("{}", 1 - set_value)).unwrap();
+
+                if optional_field_id.len() == 2 {
```

Comment:
Please extend the support to any number of fields.

```
if optional_field_ids.len() == 1 {
  // no additional check required
} else if optional_field_ids.len() >= 2) {
  // check flag value consistency
}
```

The check itself can be generated as a boolean condition for readability.
```
let cond_value_is_zero = a1.is_(some|none)() || a2.is_(some|none)() || .. aN.is_(some|none)();
let cond_value_is_one = a1.is_(none|some)() || a2.is_(none|some)() ||.. aN.is_(none|some)();
if cond_value_is_zero && cond_value_is_one {
  return Err(EncodeError::InconsistentConditionValue { .. });
}
```